programs/stake/src/stake_state.rs

```diff
@@ -213,17 +217,34 @@ fn calculate_stake_points_and_credits(
     new_vote_state: &VoteState,
     stake_history: Option<&StakeHistory>,
     inflation_point_calc_tracer: Option<impl Fn(&InflationPointCalculationEvent)>,
-) -> (u128, u64) {
+    credits_auto_rewind: bool,
+) -> (u128, u64, bool) {
+    let credits_in_stake = stake.credits_observed;
+    let credits_in_vote = new_vote_state.credits();
     // if there is no newer credits since observed, return no point
-    if new_vote_state.credits() <= stake.credits_observed {
-        if let Some(inflation_point_calc_tracer) = inflation_point_calc_tracer.as_ref() {
-            inflation_point_calc_tracer(&SkippedReason::ZeroCreditsAndReturnCurrent.into());
-        }
-        return (0, stake.credits_observed);
+    if credits_in_vote <= credits_in_stake {
+        let (referenced_credits, forced_credits_update_with_skipped_reward) = if credits_auto_rewind
+            && credits_in_vote < credits_in_stake
+        {
+            if let Some(inflation_point_calc_tracer) = inflation_point_calc_tracer.as_ref() {
+                inflation_point_calc_tracer(&SkippedReason::ZeroCreditsAndReturnRewinded.into());
+            }
+            (credits_in_vote, true)
+        } else {
```

Comment:
namely, don't force to update for `credits_in_vote == credits_in_stake` both before and after feature activation for compatibility. (add comment).

Reply:
maybe, add `// add else if credits_in_vote == credits_in_stake` for later clear guidance for post-feature-activation code cleaning.